### PR TITLE
Search for name in range of lines and add flexible sourcelinks

### LINKED ISSUE #12762: Blog post links are generated incorrectly
At the [blog page](https://dotty.epfl.ch/blog/index.html) of the Dotty website, the latest blog to date is "Scala 3.0.1-RC1 – further stabilising the compiler". The link to it is `https://dotty.epfl.ch/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md`. It should be `https://dotty.epfl.ch/blog/2021/06/07/scala3.0.1-rc1-release.html` instead. Note that in the left sidebar, the link to that blog post is generated correctly.

### PR DESCRIPTION
**Pull request by romanowski**
fixes #12762

fixes problems with sourcelink tests

### CODE REVIEW

File: docs/blog/index.html [hunk 1]
```diff
@@ -9,7 +9,7 @@ <h1>{{ page.title }}</h1>
         {% for post in site.posts %}
             <li>
                 <h2>
-                    <a href="{{ site.baseurl }}{{ post.url }}">{{ post.title }}</a>
+                    <a href="{{ post.url }}">{{ post.title }}</a>
```
Comment (BarkingBad) [inline docs/blog/index.html:12]: Was `site.baseurl` empty string?

**Comment by romanowski** [comment]
[Doc](https://scala3doc.virtuslab.com/pr-fix-sourcelinks/scala3/blog/index.html - ) generated using this PR - latest blogpost links correctly

File: project/Build.scala [hunk 2]
```diff
@@ -1244,7 +1244,12 @@ object Build {
       // TODO add versions etc.
       def srcManaged(v: String, s: String) = s"out/bootstrap/stdlib-bootstrapped/scala-$v/src_managed/main/$s-library-src"
       def scalaSrcLink(v: String, s: String) = s"-source-links:$s=github://scala/scala/v$v#src/library"
-      def dottySrcLink(v: String, s: String) = s"-source-links:$s=github://lampepfl/dotty/$v#library/src"
+      def dottySrcLink(v: String, s: String) =
+        sys.env.get("GITHUB_SHA") match {
```
Comment (BarkingBad) [inline project/Build.scala:1248]: Why do we need this `environment variables` driven logic?

File: scaladoc/test-source-links/dotty/tools/scaladoc/source-links/RemoteLinksTest.scala [hunk 3]
```diff
@@ -36,16 +37,23 @@ class RemoteLinksTest:
...
         val doc = getDocumentFromUrl(link)
-        members.foreach { (member, line) =>
+        println(s"Checking $link")
+        members.foreach { case (member, expectedLine) =>
           if !member.startsWith("given_") then // TODO: handle synthetic givens, for now we disable them from testing
-            val loc = doc.select(s"#LC$line").text
+            val toLine = expectedLine + 3
             val memberToMatch = member.replace("`", "")
-            assertTrue(s"Expected to find $memberToMatch at $link at line $line", loc.contains(memberToMatch))
+            val lineCorrectlyDefined = (expectedLine.toInt until toLine.toInt).exists{ line =>
```
Comment (BarkingBad) [inline scaladoc/test-source-links/dotty/tools/scaladoc/source-links/RemoteLinksTest.scala:50]: If I understand it correctly we look for the member in the range of lines. Was there something failing for the exact line match?

**Review by BarkingBad — APPROVED** [review]
Changes seem OK, though I have few questions.

File: docs/blog/index.html (hunk 1 above)
Comment (romanowski) [reply docs/blog/index.html:12]: yes, we do not expose such setting

File: project/Build.scala (hunk 2 above)
Comment (romanowski) [reply project/Build.scala:1248]: The problem is that we want to link to the source code that has been actually used for this PR, not the 3.0.0 sources. Before we tried to link to e.g. experimental.scala that that not exists in 3.0.0

File: scaladoc/test-source-links/dotty/tools/scaladoc/source-links/RemoteLinksTest.scala (hunk 3 above)
Comment (romanowski) [reply scaladoc/test-source-links/dotty/tools/scaladoc/source-links/RemoteLinksTest.scala:50]: yes, in was the case of name that spans in multiple lines like `asMatchable`:  https://github.com/lampepfl/dotty/blob/3.0.0/library/src/scala/compiletime/package.scala#L169-L170

**Pull request merged**